### PR TITLE
Dep updates and expansion of ruff CI

### PR DESCRIPTION
**Pull request by pvandyken**
Since ruff now supports black-compatible formatting, I decided to drop our `black` and `isort` dependencies and switch fully over to ruff. I also enabled all the ruff rule sets that I thought made sense for this repo. Notably, nothing to do with type checking (pyright handles that). Our code base was already quite conventional, so enabling most of these wasn't too burdensome, mostly a good chance to catch up on docstrings!

I left the ruleset additions sprawled across the commits to make it easier to see what rules enforced what changes. There's no intentional behavioural changes, so assuming the tests pass, I'll just wait for one review here.

### CODE REVIEW

File: snakebids/tests/strategies.py [hunk 1]
```diff
@@ -537,32 +537,43 @@ def multiselect_dicts(
...
+        )
 
 
 def _is_hashable(item: Any, /):
     try:
         hash(item)
-        return True
     except TypeError:
         return False
+    else:
```
Comment (kaitj) [inline snakebids/tests/strategies.py:566]: Does this need the `else:` here or would something like the code block work?

```python
def _is_hashable(item: Any, /):
    try:
        hash(item)
    except TypeError:
        return False
    return True
```

Also similar question for `supports_eq()`

File: snakebids/tests/test_app.py [hunk 2]
```diff
@@ -171,19 +171,20 @@ def test_runs_in_correct_mode(
         )
 
         # Prepare expected config
+        db_path = "/path/to/db"
```
Comment (kaitj) [inline snakebids/tests/test_app.py:174]: Just want to make sure this generic `db_path` won't cause any issues for the testing (or write somewhere unexpected if testing local)

**Review by kaitj — COMMENTED** [review]
Thanks for tackling this, also nice to see more support from ruff. Largely looks good to me with a few comments and questions to specific changes.

File: snakebids/tests/test_app.py (hunk 2 above)
Comment (pvandyken) [reply snakebids/tests/test_app.py:174]: All the io functions are mocked, so nothing gets written. I'll make this clear in a comment

**Review by kaitj — APPROVED** [review]

**Pull request merged**